esmvaltool/diag_scripts/fire/diagnostic_run_confire.py

```diff
@@ -0,0 +1,1230 @@
+            provenance_logger.log(filename, provenance)
+
+    # --------------------------------------------------------
+    # **Visualization: Plot Resultant Maps**
+    # --------------------------------------------------------
+    # **Load and Plot Output Data**
+    logger.info("Plotting model diagnostic outputs...")
+    figures = []
+    parameter_plot = {
+        "burnt_fraction": {
+            "vmin": 0.0,
+            "vmax": 100.0,
+        },
+        "fire_weather_control": {
+            "vmin": 0.0,
+            "vmax": 1.0,
+        },
+        "fuel_load_continuity_control": {
+            "vmin": 0.0,
+            "vmax": 1.0,
+        },
+        "stochastic_control": {
+            "vmin": 0.0,
+            "vmax": 1.0,
+        },
+    }
+    for filename in config["filenames_out"]:
+        filepath = Path(output_dir) / f"{filename}_{model_name}_{timerange}.nc"
+        fig, axes = plt.subplots(
+            nrows=1,
+            ncols=2,
+            figsize=(12, 8),
+            subplot_kw={"projection": ccrs.Robinson()},
+        )
+        axes = axes.flatten()
+        plotn = 0
+        try:
+            # Load saved NetCDF file
+            cube = iris.load_cube(filepath)
+            for pct in [5, 95]:
+                pc_cube = cube.collapsed(
+                    "realization",
+                    iris.analysis.PERCENTILE,
+                    percent=pct,
+                )[0]
+                img = iris.quickplot.pcolormesh(
+                    pc_cube,
+                    axes=axes[plotn],
+                    colorbar=False,
+                    vmin=parameter_plot[filename]["vmin"],
+                    vmax=parameter_plot[filename]["vmax"],
+                    cmap="Oranges",
+                )
+                axes[plotn].set_title(
+                    (
+                        filename.replace("_", " ").capitalize()
+                        + f" [{pct!s}th percentile]\n"
+                        + f"{model_name} "
+                        + f"({'/'.join(project)} - "
+                        + f"{'/'.join(experiment)}, "
+                        + f"{timerange})"
+                    ),
+                )
+                axes[plotn].coastlines()
+                # Add colorbar
+                fig.colorbar(
+                    img,
+                    ax=axes[plotn],
+                    orientation="vertical",
+                    extend="neither",
+                    label=filename.replace("_", " ").capitalize()
+                    + f" [{cube.units}]",
+                    fraction=0.046,
+                    pad=0.04,
+                    shrink=0.3,
+                )
+                # Iterate plot number
+                plotn = plotn + 1
+            figures.append(fig)
+
+        except ValueError as expt:
+            logger.info("Skipping %s due to error: %s", filename, expt)
+            logger.debug("Skipping %s due to error: %s", filename, expt)
+
+    return figures
```

Comment:
relatively large script: (optional) could we maybe break it into smaller bits? No worries if that's too much hassle